Sara install gem devise

saradotlog opened the pull request:
## What type of PR is this? (check all applicable)

- [x] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update


## Description of what PR does

Installs the gem Devise needed for user authentication

## Related PRs (if any)
-
-


## QA Instructions, Screenshots, Recordings



## Added tests?

- [ ] yes
- [x] no, because they aren't needed _(please include reasons for why tests aren't needed)_
- [ ] no, because I need help


## Added to documentation?

- [ ] Yes, project README
- [x] No documentation needed


## [optional] What gif best describes this PR or how it makes you feel?

![Alt-Text](GIF URL)

msarit reviewed (approved):
@saradotlog This looks good! Next steps will be to actually build out the authentication UI and place the rest of the application behind it.

Thanks for getting the Devise gem in! 👍🏾

GabrielleEaston reviewed (approved):
Pulled down the PR to review branch on my local env. 
After running bundle and rails db:migrate the App was running just fine on my localhost.
Thanks @saradotlog

saradotlog commented:
Hi @aritamana I've been looking further the devise documentation and I'm thinking, wouldn't be better if I replace the class name used for the application’s users, and call it User? I didn't realise when I ran the command `rails generate devise MODEL` and I called it MODEL instead of User

msarit commented:
I agree @saradotlog. Please go ahead and let me know if you need help. Sorry I missed that. 😅

ncbui reviewed (dismissed):
Pulled into my local machine for PR. Ran `bundle install` to install gems and the app works fine. Great job @saradotlog !

saradotlog commented:
Hi all! I updated the code. Removed the model class and generated a User one

Can you please review it again? Thanks! @msarit @GabrielleEaston @Timilehin08 @cclics @ncbui

msarit commented:
@saradotlog There are conflicts to resolve, Sara. Please update your branch with upstream and resolve conflicts.

msarit commented on app/controllers/shifts_controller.rb:
```diff
@@ -1,4 +1,5 @@
 class ShiftsController < ApplicationController
+  before_action :authenticate_user!, only: %i[ index show edit update destroy ]
```
@saradotlog I added this `before_action` to the `Shifts` controller, so that the app checks for if the user is logged in before executing any of the methods in the array `%i[ index show edit update destroy ]`

msarit commented on app/views/welcome/index.html.erb:
```diff
@@ -1,4 +1,14 @@
 <h1>ShiftWork App</h1>
+<p>
```
@saradotlog I added appropriate 'Sign In' and 'Log In' links to the Welcome page

msarit commented:
@ncbui @GabrielleEaston Please re-review so I can merge! 👍🏾 🙏🏾

GabrielleEaston reviewed (approved):
After struggling with Postgres and denied permission to the database (for a few hours 🥲), I successfully got to the login, signup page. Great job, @saradotlog

ncbui commented:
I had a bit of trouble migrating db but having got it going, too. Great work @saradotlog 

I am able to signup, login, change my password, change my email, and cancel an account. 
When I try to log in to a nonexistent account, I get an appropriate error.
When I try to log in to an existing account with the wrong password, I get an appropriate error
